tools/kagami/src/validator.rs

```diff
@@ -6,46 +6,63 @@ use swarm::AbsolutePath;
 
 use super::*;
 
-#[derive(ClapArgs, Debug, Clone, Copy)]
-pub struct Args;
+#[derive(ClapArgs, Debug, Clone)]
+pub struct Args {
+    /// Directory path to clone Iroha sources to. Only
+    /// used in case when `kagami` is run as a separate
+    /// binary. A temporary directory is created for this
+    /// purpose if this option is not provided but cloning
+    /// is still needed.
+    #[clap(long)]
+    clone_dir: Option<PathBuf>,
+}
 
 impl<T: Write> RunArgs<T> for Args {
     fn run(self, writer: &mut BufWriter<T>) -> Outcome {
-        let path = compute_validator_path()?;
+        let clone_dir = match self.clone_dir {
+            Some(dir) => dir,
+            None => tempfile::tempdir()
+                .wrap_err("Failed to generate a tempdir for validator sources")?
+                .into_path(),
+        };
+        let path = compute_validator_path(clone_dir.as_path())?;
         writer
             .write_all(&construct_validator(path)?)
             .wrap_err("Failed to write wasm validator into the buffer.")
     }
 }
-pub fn compute_validator_path() -> Result<PathBuf> {
+
+pub fn compute_validator_path(out_dir: impl AsRef<Path>) -> Result<PathBuf> {
```

Comment:
Please, add some doc comments to this function.

I see this code for the first time